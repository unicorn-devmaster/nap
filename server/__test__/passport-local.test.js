/* eslint-env jest */

const config = require('../config')

describe('passport-local', () => {
  it('should throw error for empty email', async () => {
    const { willValidateEmail } = require('../passport-local')
    await willValidateEmail().catch(err => {
      expect(() => { throw err }).toThrow('Required : email')
    })
  })

  it('should throw error for invalid email', async () => {
    const { willValidateEmail } = require('../passport-local')
    await willValidateEmail('').catch(err => {
      expect(() => { throw err }).toThrow('Invalid email')
    })
  })

  it('should be true for valid email', async () => {
    const { willValidateEmail } = require('../passport-local')
    expect(await willValidateEmail('foo@bar.com')).toMatchSnapshot()
  })

  it('should throw error for empty password', async () => {
    const { willValidatePassword } = require('../passport-local')
    await willValidatePassword().catch(err => {
      expect(() => { throw err }).toThrow('Required : password')
    })
  })

  it('should throw error for invalid password', async () => {
    const { willValidatePassword } = require('../passport-local')
    const { PASSWORD_LENGTH_ERROR } = require('../errors')
    await willValidatePassword('foo').catch(err => {
      expect(() => { throw err }).toThrow(PASSWORD_LENGTH_ERROR.message)
    })
  })

  it('should be true for valid passowrd', async () => {
    const { willValidatePassword } = require('../passport-local')
    expect(await willValidatePassword('foofoobarbar')).toMatchSnapshot()
  })
})
