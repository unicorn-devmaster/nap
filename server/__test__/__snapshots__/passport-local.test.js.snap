// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passport-local should be true for valid email 1`] = `true`;

exports[`passport-local should be true for valid passowrd 1`] = `true`;
